packages/api/src/UserModule/UserModule.service.ts

```diff
@@ -65,36 +73,48 @@ export class UserModuleService {
       throw new BadRequestException('Incorrect lesson submission');
     }
 
-
     // Check ALL answers, and return false if even one is wrong
-    const allCorrect = this.cms.findLessonById(moduleId)!.lesson.questions.every(q => {
-      // Memory game is not checked as it's client only
-      if (q.type === QuestionType.memory) return true;
-
-      if (!answers[q.id]) return false; // answer not submitted
-
-      const correct = this.cms.checkAnswer(q.id, answers[q.id]).every(a => a);
-
-      if (config.get('env') === 'development' && !correct) {
-        // eslint-disable-next-line no-console
-        console.log(`Incorrect question answer ${q.id}. Expected ${JSON.stringify(q, null, 2)}`);
-      }
-
-      return correct;
-    });
+    const allCorrect = this.cms
+      .findLessonById(moduleId)!
+      .lesson.questions.every(q => {
+        // Memory game is not checked as it's client only
+        if (q.type === QuestionType.memory) return true;
+
+        if (!answers[q.id]) return false; // answer not submitted
+
+        const correct = this.cms
+          .checkAnswer(q.id, answers[q.id])
+          .every(a => a);
+
+        if (config.get('env') === 'development' && !correct) {
+          // eslint-disable-next-line no-console
+          console.log(
+            `Incorrect question answer ${q.id}. Expected ${JSON.stringify(
+              q,
+              null,
+              2
+            )}`
+          );
+        }
+
+        return correct;
+      });
 
     if (!allCorrect) return false;
 
     um.completedAt = new Date();
+
     await um.save();
 
+    await this.userService.incrementStreak(userId);
```

Comment:
Return value of `incrementStreak` is never used, so it should not return anything; it should simply do the operation of incrementing streak.

Reply:
Sorry about that. This has been updated.